### PR TITLE
Help Text pop up

### PR DESCRIPTION
**Pull request by elenirotsides**
- created PopUp component
- edited strings so that we can dynamically render the helptext
- added PopUp component to submit rating page

closes https://github.com/Peer-Stevens/peer/issues/278

Did not style things fully (this should be done in another issue), I only added the component to the submit rating page. Each helptext item is being dynamically rendered along with each attribute on the page.

Decided to go with a modal instead of a tooltip bc a tooltip feels more web based to me. Also, I think a pop up would better suit our users than a tooltip.

### CODE REVIEW

File: src/components/PopUp.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+			<Modal
+				animationType="none"
+				// TODO: setting 'transparent' to false will make the background disappear; we may want this for our users
+				// so that there are less distractions and they can focus on a single element. Test it out and lmk what you guys think
+				transparent={true}
+				visible={modalVisible}
+				onRequestClose={() => {
+					Alert.alert("Modal has been closed.");
+					setModalVisible(!modalVisible);
+				}}
```
Comment (carpenterd777) [inline src/components/PopUp.tsx:42]: haven't been able to load the app because I'm on campus, but I don't think we should use words like "modal" with our users: those words are for us, and they probably won't understand what they mean. in fact, I don't think that an alert is even necessary when the dialogue box is closed.

File: src/components/PopUp.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+			>
+				<View style={styles.centeredView}>
+					<View style={styles.modalView}>
+						{children}
+						<Button
+							onPress={() => setModalVisible(!modalVisible)}
+							accessibilityLabel="Close Pop Up"
+							text="Close Pop Up"
```
Comment (carpenterd777) [inline src/components/PopUp.tsx:50]: nit
```suggestion
							accessibilityLabel="Close pop Up"
							text="Close pop Up"
```

File: src/components/PopUp.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	accessibilityLabel,
+	accessibilityHint,
+	style,
+	children,
+}) => {
+	const [modalVisible, setModalVisible] = useState(false);
+	return (
+		<View
+			accessibilityLabel="A Pop Up"
+			accessibilityHint="A Pop Up"
```
Comment (carpenterd777) [inline src/components/PopUp.tsx:25]: I think we should put some more than just "a pop up" . What's the pop-up here for?

File: src/screens/SubmitRatingScreen.tsx [hunk 4]
```diff
@@ -82,18 +84,33 @@ const SubmitRatingScreen: React.FC<SubmitRatingScreenProps> = ({
...
+							accessibilityLabel={
+								"A button that triggers a pop up to open. This will explain what " +
+								attribute.type +
+								" is"
+							}
+							accessibilityHint={
+								"A button that triggers a pop up to open. This will explain what " +
+								attribute.type +
+								" is"
+							}
```
Comment (carpenterd777) [inline src/screens/SubmitRatingScreen.tsx:102]: can we move all strings and stringbuilding functions to `strings.ts`?

**Review by carpenterd777 — CHANGES_REQUESTED** [review]

File: src/components/PopUp.tsx (hunk 3 above)
Comment (elenirotsides) [reply src/components/PopUp.tsx:25]: I designed this component to serve a broader purpose. We can use this for help text, but we can also use it in other places as well. I wasn't really sure what to put here since my idea was to make this more of a general component. I can make these fields props that we pass in if we want these to be more descriptive, lmk what you think @carpenterd777

Comment (carpenterd777) [reply src/components/PopUp.tsx:25]: yeah, for a reusable component I would def prefer making the accessibility fields props @elenirotsides

**Comment by carpenterd777** [comment]
just took a look at the actual visuals of the pop-up, and I think we should modernize the UI a little bit. instead of creating a dialogue in the center of the screen, what if a view slid up from the bottom of the screen?

**Comment by elenirotsides** [comment]
> just took a look at the actual visuals of the pop-up, and I think we should modernize the UI a little bit. instead of creating a dialogue in the center of the screen, what if a view slid up from the bottom of the screen?

This what I was originally shooting for, but I got stuck. I'm not quite sure how to achieve this at the moment, but I'm gonna keep researching.

**Comment by carpenterd777** [comment]
> > just took a look at the actual visuals of the pop-up, and I think we should modernize the UI a little bit. instead of creating a dialogue in the center of the screen, what if a view slid up from the bottom of the screen?
> 
> This what I was originally shooting for, but I got stuck. I'm not quite sure how to achieve this at the moment, but I'm gonna keep researching.

that's fair. it might not be as important as I think, but I just wanted us to start with the feel (sans look) of the component that we plan to build off of so that we don't have to start from scratch later. thoughts @aej11a ?

File: src/components/PopUp.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+				{
+					backgroundColor: PRIMARY_COLOR,
+				},
+				style
+			)}
+		>
+			<Modal
+				animationType="none"
+				// TODO: setting 'transparent' to false will make the background disappear; we may want this for our users
+				// so that there are less distractions and they can focus on a single element. Test it out and lmk what you guys think
```
Comment (carpenterd777) [inline src/components/PopUp.tsx:36]: tried it out, and I'm not too big a fan on the background disappearing, I think it looks too much like a glitch with the app or something. if we could get the background to be darkened so that the background elements are only a little visible but are still clearly there, I think that would be more effective at letting the user know that the background elements are not in focus right now.

File: src/util/strings.ts [hunk 6]
```diff
@@ -50,6 +50,39 @@ export const getIncrementRatingButtonLabel = (
...
+
+	switch (propsType) {
+		case "buttonAccessibility":
+			displayString = `A button that triggers a pop up to open. This will explain what ${attribute} is.`;
+			break;
+		case "modalAccessibility":
+			displayString = `A pop up that explains ${attribute}`;
+			break;
+		case "closeButtonAccessibilityLabel":
+			displayString = `Close ${attribute} pop up`;
```
Comment (aej11a) [inline src/util/strings.ts:77]: Listening to this, i think it's too wordy - they can only have one popup open at-a-time anyway.
Just "close this pop up" is probably good

File: src/util/strings.ts [hunk 7]
```diff
@@ -50,6 +50,39 @@ export const getIncrementRatingButtonLabel = (
...
+		| "buttonAccessibility"
+		| "text"
+		| "modalAccessibility"
+		| "closeButtonAccessibilityLabel"
+): string => {
+	let displayString!: string;
+
+	switch (propsType) {
+		case "buttonAccessibility":
+			displayString = `A button that triggers a pop up to open. This will explain what ${attribute} is.`;
```
Comment (aej11a) [inline src/util/strings.ts:71]: less techy more consumer-oriented. this isn't a description, it should be concise and clear and CTA-ish.

ex: Press this button to find out more about the ${attribute} rating.

**Review by aej11a — CHANGES_REQUESTED** [review]

File: src/util/strings.ts (hunk 6 above)
Comment (elenirotsides) [reply src/util/strings.ts:77]: That's a good point

File: src/components/PopUp.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
+import React, { useState } from "react";
+import { View, StyleSheet, ViewStyle } from "react-native";
+import type { StyleProp } from "react-native";
+import { PRIMARY_COLOR, TEXT_COLOR } from "../util/colors";
+import { Button } from "./Button";
+import Modal from "react-native-modal";
+
+export interface ModalProps {
```
Comment (carpenterd777) [inline src/components/PopUp.tsx:8]: nit
```suggestion
export interface PopUpProps {
```

File: src/components/PopUp.tsx [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+	closeButtonText,
+	closeButtonAccessibilityLabel,
+	style,
+	children,
+}) => {
+	const [modalVisible, setModalVisible] = useState(false);
+	return (
+		<View
+			accessibilityLabel={modalAccessibility}
+			accessibilityHint={modalAccessibility}
```
Comment (carpenterd777) [inline src/components/PopUp.tsx:32]:
```suggestion
```
having this twice will cause the element to repeat its text twice on Android

**Review by carpenterd777 — CHANGES_REQUESTED** [review]

File: src/components/PopUp.tsx (hunk 9 above)
Comment (elenirotsides) [reply src/components/PopUp.tsx:32]: ahhh, I did not know that! So `accessibilityLabel` is sufficient then? What's the difference? @carpenterd777

**Review by carpenterd777 — APPROVED** [review]
